### PR TITLE
Documentation generation in net requests and client

### PR DESCRIPTION
**Pull request by RachelTucker**

### CODE REVIEW

**Comment by RachelTucker** [comment]
Updated

File: ds3-autogen-java/src/test/java/com/spectralogic/ds3autogen/java/utils/TestGeneratedCode.java [hunk 1]
```diff
@@ -50,12 +50,12 @@ public TestGeneratedCode(
...
     public void generateCode(
             final FileUtils fileUtils,
-            final String inputFileName) throws IOException, ParserException, ResponseTypeNotFoundException, TypeRenamingConflictException, TemplateModelException {
+            final String inputFileName) throws IOException, TemplateModelException {
         final Ds3SpecParser parser = new Ds3SpecParserImpl();
         final Ds3ApiSpec spec = parser.getSpec(TestGeneratedCode.class.getResourceAsStream(inputFileName));
         final CodeGenerator codeGenerator = new JavaCodeGenerator();
 
-        codeGenerator.generate(spec, fileUtils, Paths.get("."));
+        codeGenerator.generate(spec, fileUtils, Paths.get("."), null);
```
Comment (DenverM80) [inline ds3-autogen-java/src/test/java/com/spectralogic/ds3autogen/java/utils/TestGeneratedCode.java]: can you overload the generate method, so you don't have to pass null here?

Comment (rpmoore) [inline ds3-autogen-java/src/test/java/com/spectralogic/ds3autogen/java/utils/TestGeneratedCode.java]: +1 this

Comment (RachelTucker) [inline ds3-autogen-java/src/test/java/com/spectralogic/ds3autogen/java/utils/TestGeneratedCode.java]: I could, but for the most part, I want to specify a Ds3DocSpec. The null was just a quick way to ensure current tests continued passing. In my next pull request, I've created an empty Ds3DocSpec, so I was going to use that instead in these instances.

Comment (rpmoore) [inline ds3-autogen-java/src/test/java/com/spectralogic/ds3autogen/java/utils/TestGeneratedCode.java]: I like the empty Ds3DocSpec even more, so we'll go with that in the next pull request.

File: ds3-autogen-net/src/main/java/com/spectralogic/ds3autogen/net/generators/clientmodels/BaseClientGenerator.java [hunk 2]
```diff
@@ -113,11 +122,9 @@ protected static String getHttpStatusCode(final ImmutableList<Ds3ResponseCode> r
...
         }
         final ImmutableList.Builder<Integer> builder = ImmutableList.builder();
-        for (final Ds3ResponseCode responseCode : responseCodes) {
-            if (ResponsePayloadUtil.isNonErrorCode(responseCode.getCode())) {
-                builder.add(responseCode.getCode());
-            }
-        }
+        responseCodes.stream()
+                .filter(responseCode -> ResponsePayloadUtil.isNonErrorCode(responseCode.getCode()))
+                .forEach(responseCode -> builder.add(responseCode.getCode()));
```
Comment (DenverM80) [inline ds3-autogen-net/src/main/java/com/spectralogic/ds3autogen/net/generators/clientmodels/BaseClientGenerator.java]: :+1:

File: ds3-autogen-c/src/test/java/com/spectralogic/ds3autogen/c/CCodeGeneratorSpectraS3Requests_Test.java [hunk 3]
```diff
@@ -46,7 +43,7 @@
 public class CCodeGeneratorSpectraS3Requests_Test {
 
     @Test
-    public void testGenerateSpectraS3DeleteBucketRequest() throws IOException, ParserException, ResponseTypeNotFoundException, TypeRenamingConflictException, ParseException, TemplateModelException {
+    public void testGenerateSpectraS3DeleteBucketRequest() throws IOException, ParseException, TemplateModelException {
```
Comment (rpmoore) [inline ds3-autogen-c/src/test/java/com/spectralogic/ds3autogen/c/CCodeGeneratorSpectraS3Requests_Test.java]: Wasn't ParseException made into a RuntimeException?

Comment (RachelTucker) [inline ds3-autogen-c/src/test/java/com/spectralogic/ds3autogen/c/CCodeGeneratorSpectraS3Requests_Test.java]: Missed one!

Comment (RachelTucker) [inline ds3-autogen-c/src/test/java/com/spectralogic/ds3autogen/c/CCodeGeneratorSpectraS3Requests_Test.java]: Nm. ParserException was changed to runtime, but ParseException is not one of ours.

Comment (rpmoore) [inline ds3-autogen-c/src/test/java/com/spectralogic/ds3autogen/c/CCodeGeneratorSpectraS3Requests_Test.java]: Ah, it's apart of the std lib.

File: ds3-autogen-net/src/main/java/com/spectralogic/ds3autogen/net/generators/clientmodels/BaseClientGenerator.java (hunk 2 above)
Comment (rpmoore) [inline ds3-autogen-net/src/main/java/com/spectralogic/ds3autogen/net/generators/clientmodels/BaseClientGenerator.java]: This should use a collector instead to create the list.

**Pull request merged**